typespec/Applications.Core/containers.tsp

```diff
@@ -240,7 +240,7 @@ model Container {
   imagePullPolicy?: ImagePullPolicy;
 
   @doc("environment")
-  env?: Record<string>;
+  env?: Record<EnvironmentVariable>;
```

Comment:
Note to reviewers, we have to make a breaking change to add this feature. We'll discuss the API changes in design review.

Reply:
Example of the updated usage here: https://github.com/radius-project/radius/pull/7744/files#diff-19f5067c27bb52be17b49eb73842441f50662b7ef8317d161ed0e6863a26f0d0R29